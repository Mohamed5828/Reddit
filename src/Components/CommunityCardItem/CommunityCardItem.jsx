import {useState} from "react";
import logo from "Assets/Images/logo.png";
import {Container} from "./CommunityCardItem.styled";
import {Link} from "react-router-dom";
import {FaAngleUp, FaAngleDown} from "react-icons/fa";
import {useNavigate} from "react-router-dom";

/**
 * Component that  shows the names of communities up-to-date.
 * @returns {Component.React}
 */
const CommunityCardItem = ({communityId, community}) => {
    const [btnContent, setBtnContent] = useState("Join");

    const clickHandler = (e) => {
        e.preventDefault();
        if (btnContent === "Join") {
            setBtnContent("Joined");
        } else {
            setBtnContent("Join");
        }
    }

    const mouseEnterHandler = () => {
        if (btnContent === "Joined") {
            setBtnContent("Leave");
        }
    }

    const MouseLeaveHandler = () => {
        if (btnContent === "Leave") {
            setBtnContent("Joined");
        }
    }
    const navigate = useNavigate();

    return (
        <Container>
            <Link to={"/subreddit"}>
                <div className="item">
                    <span className={'num'}>{communityId}</span>
                    <span className='caret'><FaAngleUp/></span>
                    <img src={logo} alt=""/>
                    <div  className="info">
                        <div  className="info2">
                            <h6 >r/{community}</h6>

                        </div>
                    </div>
                    <div className="button">
                        <button
                            onClick={clickHandler}
                            onMouseEnter={mouseEnterHandler}
                            onMouseLeave={MouseLeaveHandler}>
                            {btnContent}
                        </button>
                    </div>
                </div>
            </Link>
        </Container>
    );
};

export default CommunityCardItem;