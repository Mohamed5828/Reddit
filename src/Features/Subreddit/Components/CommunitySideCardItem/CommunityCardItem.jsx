import logo from "../../Assets/images/logo.svg";
import {Container,
CommunityA,
CommunityIndex,
Arrow,
CommunityImg,
TitleParagraph,
} from "./CommunityCardItem.styled";

/**
 * @Component
 * @param {string} title - Subreddit Title
 * @param {number} index - Subreddit index among fetched data to index the list
 * Component that  shows the names of communities up-to-date.
 * @returns {Component.React}
 */
const CommunityCardItem = ({index, title}) => {

    return (
        <Container>
            <CommunityA to={`/subreddit`}>
                <CommunityIndex>{index}</CommunityIndex>
                <Arrow />
                 <CommunityImg
                    src={logo}
                    alt="logo"
                ></CommunityImg>
                <TitleParagraph>{title}</TitleParagraph> 
            </CommunityA>
        </Container>
    );
};

export default CommunityCardItem;