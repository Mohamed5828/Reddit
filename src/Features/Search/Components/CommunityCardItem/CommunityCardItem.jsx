import { useState } from "react";
import { Link } from "react-router-dom";
// import logo from "../../Assets/download.jpg";
import { Container } from "./CommunityCardItem.styled";
// Import api
import axios from "API/axios";
import useFetchFunction from "Hooks/useFetchFunction";
/**
 * Component that contains the CommunityCardItem and manage the state of the button join.
 *
 * @Component
 * @param {String} communityIcon - The image of the CommunityCardItem
 * @param {String} communityID - The ID of the CommunityCardItem
 * @param {String} communityDescription - The Description of the CommunityCardItem
 * @param {number} membersCount - The membersCount of the CommunityCardItem
 * @param {String} communityName - The name of the CommunityCardItem
 * @param {object} isJoined - it is object indicating whether the user is joining this community
 * @returns {React.Component}
 */
const CommunityCardItem = ({
  communityIcon,
  communityID,
  communityDescription,
  communityName,
  membersCount,
  isJoined,
}) => {
  const [joinRes, errorJoin, joinLoading, fetchFunction] = useFetchFunction();
  const [isJoinedstate, setisJoined] = useState(false);
  const joinCommunity = (btnState) => {
    fetchFunction({
      axiosInstance: axios,
      method: "POST",
      url: "http://localhost:8000/Join",
      requestConfig: {
        headers: {
          "Content-Language": "en-US",
        },
        data: {
          action: !btnState ? "unsub" : "sub",
          sr_name: `${communityName}`,
        },
      },
    });
  };
  const initialState = `${isJoined !== undefined ? "Joined" : "Join"}`;
  // the state of the buuton
  const [btnContent, setBtnContent] = useState(
    isJoined !== undefined ? "Joined" : "Join"
  );

  /**
   * it is the function that handle the state of the button when click on it.
   *
   * @param {Object} e it is an object which i can preventDefault and not redirect to the link
   */

  const clickHandler = (e) => {
    e.preventDefault();
    if (btnContent === "Join") {
      setBtnContent("Joined");
      setisJoined(true);
    } else {
      setBtnContent("Join");
      setisJoined(false);
    }
    let btnState;
    if (btnContent === "Join") {
      btnState = true;
    } else {
      btnState = false;
    }
    joinCommunity(btnState);
  };
  /**
   * it is the function that handle the state of the button when mouseEnter on it.
   *
   */
  function mouseEnterHandler() {
    if (btnContent === "Joined") {
      setBtnContent("Leave");
    }
  }
  /**
   * it is the function that handle the state of the button when MouseLeave on it.
   *
   */
  function MouseLeaveHandler() {
    if (btnContent === "Leave") {
      setBtnContent("Joined");
    }
  }
  return (
    <Container>
      <Link href="#">
        <div className="item">
          <img src={require(`../../Assets/${communityIcon}`)} alt="" />
          <div className="info">
            <div className="info2">
              <h6>r/{communityName}</h6>
              <p>{membersCount}m Members</p>
            </div>
          </div>
          <div className="button">
            <button
              onClick={clickHandler}
              onMouseEnter={mouseEnterHandler}
              onMouseLeave={MouseLeaveHandler}
              className="join"
            >
              {btnContent}
            </button>
          </div>
        </div>
      </Link>
    </Container>
  );
};

export default CommunityCardItem;
