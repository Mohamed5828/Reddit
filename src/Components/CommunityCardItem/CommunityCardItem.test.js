import {shallow} from 'enzyme';

// import Components
import CommunityCardItem from "./CommunityCardItem";
import { BrowserRouter, Routes } from "react-router-dom";

describe("Community Card Item Component", () => {
    it("this is a test for Community Card Item Component", () => {
        expect(
          shallow(
            <BrowserRouter>
              <Routes>
                <CommunityCardItem />
              </Routes>
            </BrowserRouter>
          )
        ).toMatchSnapshot();

    });
});