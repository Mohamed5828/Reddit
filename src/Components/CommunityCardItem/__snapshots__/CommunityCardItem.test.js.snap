// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Card Item Component this is a test for Community Card Item Component 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes>
    <CommunityCardItem />
  </Routes>
</Router>
`;
